src/main/java/com/iridium/iridiumskyblock/managers/tablemanagers/UserTableManager.java

```diff
@@ -14,49 +14,49 @@
  */
 public class UserTableManager extends TableManager<User, Integer> {
 
-    // A list of users sorted by island id for binary search
-    private final SortedList<User> userIslandIndex;
+    private final LinkedHashMap<UUID, User> userIslandMap = new LinkedHashMap<>();
 
     public UserTableManager(ConnectionSource connectionSource) throws SQLException {
         super(connectionSource, User.class, Comparator.comparing(User::getUuid));
-        this.userIslandIndex = new SortedList<>(Comparator.comparing(user -> user.getIsland().map(Island::getId).orElse(0)));
-        this.userIslandIndex.addAll(getEntries());
-        sort();
-    }
-
-    /**
-     * Sort the list of entries by UUID
-     */
-    public void sort() {
-        getEntries().sort(Comparator.comparing(User::getUuid));
-        userIslandIndex.sort(Comparator.comparing(user -> user.getIsland().map(Island::getId).orElse(0)));
+        int size = getEntries().size();
+        for (int i = 0; i < size; i++) {
+            User user = getEntries().get(i);
+            userIslandMap.put(user.getUuid(), user);
+        }
     }
 
     /**
-     * Add an item to the list whilst maintaining sorted list
+     * Add an item to the list
      *
      * @param user The item we are adding
      */
     public void addEntry(User user) {
-        getEntries().add(user);
-        userIslandIndex.add(user);
+        if (userIslandMap.get(user.getUuid()) != null) return;
+        userIslandMap.put(user.getUuid(), user);
     }
 
     /**
-     * Puts the user in the correct order of userIslandIndex
+     * Puts the user in userIslandMap
      *
      * @param user The specified User
      */
     public void resortIsland(User user) {
-        userIslandIndex.remove(user);
-        int islandIndex = Collections.binarySearch(userIslandIndex, user, Comparator.comparing(u -> u.getIsland().map(Island::getId).orElse(0)));
-        userIslandIndex.add(islandIndex < 0 ? -(islandIndex + 1) : islandIndex, user);
+        if (userIslandMap.get(user.getUuid()) != null) {
```

Comment:
This method was origionally used to sort the User's via the IslandID, so we can get a list of all members in an island.

Reply:
But now, its not list, its linkedhashmap